packages/aws-cdk-lib/aws-s3/test/rules.test.ts

```diff
@@ -329,9 +330,167 @@ describe('rules', () => {
         Rules: [{
           ObjectSizeLessThan: 0,
           ObjectSizeGreaterThan: 0,
+          ExpirationInDays: 30,
           Status: 'Enabled',
         }],
       },
     });
   });
+
+  describe('required properties for rules', () => {
+    test('throw if there is a rule doesn\'t have required properties', () => {
+      const stack = new Stack();
+      new Bucket(stack, 'MyBucket', {
+        lifecycleRules: [
+          {
+            objectSizeLessThan: 300000,
+            objectSizeGreaterThan: 200000,
+          },
+        ],
+      });
+      expect(() => {
+        Template.fromStack(stack);
```

Comment:
An error occurs in synthesize phase because rendering is done as Lazy:

https://github.com/aws/aws-cdk/blob/v2.162.1/packages/aws-cdk-lib/aws-s3/lib/bucket.ts#L1941